flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/MatchCodeGenerator.scala

```diff
@@ -131,47 +132,39 @@ class MatchCodeGenerator(
     : GeneratedFunction[F, T] = {
```

Comment:
We can already reuse `FunctionCodeGenerator.generateFunction()` as IterativeCondition and Pattern(Flat)SelectFunction already support rich interfaces.

Reply:
There is one additional obstacle right now. The `reusePatternLists()` call, I would remove the comment and leave it separated, as there might be some additional differences in the future. What do you think?

Reply:
We can add something like `reusablePrepareInputStatements` in CodeGenerator and  place it before `reuseInputUnboxingCode` when generating the function. Then we can make use of `reusablePrepareInputStatements` to construct the `reusePatternLists`. But I'm fine with the current implementation as `reusablePrepareInputStatements` is currently only useful for CEP.